### ISSUE TITLE
Changing AKS node count leads to error

### ISSUE THREAD

**Issue by 4c74356b41** [pulumi/pulumi#2511]
Deploying the below code works, changing node count and trying to update fails:

error log:
```
  pulumi:pulumi:Stack (azure-py-aks-aks-repro):
    panic: interface conversion: interface {} is nil, not map[string]interface {}
    goroutine 83 [running]:
    github.com/pulumi/pulumi-azure/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm.expandKubernetesClusterAddonProfiles(0xc000919340, 0xc0006a2740)
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_kubernetes_cluster.go:752 +0x946
    github.com/pulumi/pulumi-azure/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmKubernetesClusterCreateUpdate(0xc000919340, 0x2796860, 0xc00058e000, 0x24, 0x4515e20)
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_kubernetes_cluster.go:538 +0x423
    github.com/pulumi/pulumi-azure/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc0003e8e70, 0xc0003184b0, 0xc00079b400, 0x2796860, 0xc00058e000, 0xc00079b401, 0x0, 0x0)
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:231 +0x257
    github.com/pulumi/pulumi-azure/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc000675340, 0xc000b8d820, 0xc0003184b0, 0xc00079b400, 0xc00079b400, 0x0, 0x0)
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:283 +0xa3
    github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi-terraform/pkg/tfbridge.(*Provider).Update(0xc000686e10, 0x2b7f960, 0xc0007c7860, 0xc000318280, 0xc000686e10, 0x1, 0x1)
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi-terraform/pkg/tfbridge/provider.go:687 +0x77d
    github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi/sdk/proto/go._ResourceProvider_Update_Handler.func1(0x2b7f960, 0xc0007c7860, 0x26df000, 0xc000318280, 0x2b7f960, 0xc0007c7860, 0x2b98be0, 0x45148d0)
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi/sdk/proto/go/provider.pb.go:1265 +0x8d
    github.com/pulumi/pulumi-azure/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1(0x2b7f960, 0xc0007c7860, 0x26df000, 0xc000318280, 0xc000297160, 0xc0002971a0, 0x0, 0x0, 0x2b6ac00, 0xc0002dcbc0)
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc/server.go:57 +0x2b7
    github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi/sdk/proto/go._ResourceProvider_Update_Handler(0x272bc80, 0xc000686e10, 0x2b7f960, 0xc0007c6390, 0xc000318230, 0xc0002b8040, 0x0, 0x0, 0xc000630000, 0x5565)
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi/sdk/proto/go/provider.pb.go:1267 +0x15f
    github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc000539080, 0x2b96360, 0xc000075500, 0xc0002e0700, 0xc0006c4120, 0x44e99d0, 0x0, 0x0, 0x0)
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc/server.go:966 +0x4a9
    github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc.(*Server).handleStream(0xc000539080, 0x2b96360, 0xc000075500, 0xc0002e0700, 0x0)
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc/server.go:1245 +0xd5f
    github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000598020, 0xc000539080, 0x2b96360, 0xc000075500, 0xc0002e0700)
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc/server.go:685 +0xa6
    created by github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
        /home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc/server.go:683 +0xa8
```

Configuration
```
import pulumi
from pulumi import ResourceOptions
from pulumi_azure.core import ResourceGroup
from pulumi_azure.role import Assignment
from pulumi_azure.ad import Application, ServicePrincipal, ServicePrincipalPassword
from pulumi_azure.containerservice import KubernetesCluster, Registry
from pulumi_azure.network import VirtualNetwork, Subnet

config = pulumi.Config('azure-py-aks')
PREFIX = config.get('prefix') or 'replaceme'
PASSWORD = config.get('password') or 'DEf34tg!rg35y#F23'
SSHKEY = config.get('sshkey') or 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCxinIAIDDCradZPAgX5GzBLv00u4rigOLUbU00E44FrfMTqu5wXiejJ4ycSb1bI+//ZNgaB2UYRbPL7A9OUKY+K4sX5O84Q6DPMjo/90IANHVTLf3xTaSc7hpvXOtIjJTJeiamxClgnTAcR55RV/j9/Wptxa8GGcRmRCcSmJUkx5AZTFI+s8aF0W3aeHHRw7TxNKBuwrX7FDcHyGKvdkFg4OP863Xe5hp5ql1C3XibmCOp1CMPIU2hCmGOy1LGbOf/Pa+QKAdtUSrPNK/jBWvPWo0k02Ii0JtMAdlpVqnJc3czNIp5gEqZCRCGEdkb/kZnJiMRZhmLBYnC8tiMxvZj core@k8s'
LOCATION = config.get('location') or 'westeurope'

# create Azure AD Application for AKS
app = Application(
    'aks-app',
    name=PREFIX + 'aks-app'
)

# create service principal for the application so AKS can act on behalf of the application
sp = ServicePrincipal(
    'aks-sp',
    application_id=app.application_id
)

# create service principal password
sppwd = ServicePrincipalPassword(
    'aks-sp-pwd',
    service_principal_id=sp.id,
    end_date='2025-01-01T01:02:03Z',
    value=PASSWORD
)

rg = ResourceGroup(
    'rg',
    name=PREFIX + 'rg',
    location=LOCATION
)

vnet = VirtualNetwork(
    'vnet',
    name=PREFIX + 'vnet',
    location=rg.location,
    resource_group_name=rg.name,
    address_spaces=['10.0.0.0/8']
)

subnet = Subnet(
    'subnet',
    name=PREFIX + 'subnet',
    resource_group_name=rg.name,
    address_prefix='10.0.0.0/23',
    virtual_network_name=vnet.name
)

# create Azure Container Registry to store images in
acr = Registry(
    'acr',
    name=PREFIX + 'acr',
    location=rg.location,
    resource_group_name=rg.name,
    sku="basic"
)

# assignments are needed for AKS to be able to interact with those resources
acr_assignment = Assignment(
    'acr-permissions',
    principal_id=sp.id,
    role_definition_name='AcrPull',
    scope=acr.id
)

subnet_assignment = Assignment(
    'subnet-permissions',
    principal_id=sp.id,
    role_definition_name='Network Contributor',
    scope=subnet.id
)

aks = KubernetesCluster(
    'aks',
    name=PREFIX + 'aks',
    location=rg.location,
    resource_group_name=rg.name,
    kubernetes_version="1.12.5",
    dns_prefix="dns",
    agent_pool_profile=(
        {
            "name": "type1",
            "count": 3,
            "vmSize": "Standard_B2ms",
            "osType": "Linux",
            "maxPods": 110,
            "vnet_subnet_id": subnet.id
        }
    ),
    linux_profile=(
        {
            "adminUsername": "azureuser",
            "ssh_key": [
                {
                    "keyData": SSHKEY
                }
            ]
        }
    ),
    service_principal={
        "clientId": app.application_id,
        "clientSecret": sppwd.value
    },
    role_based_access_control={
        "enabled": "true"
    },
    network_profile=(
        {
            "networkPlugin": "azure",
            "serviceCidr": "10.10.0.0/16",
            "dns_service_ip": "10.10.0.10",
            "dockerBridgeCidr": "172.17.0.1/16"
        }
    ), __opts__=ResourceOptions(depends_on=[acr_assignment, subnet_assignment])
)
```

**Comment by jen20** [comment]
I can reproduce this using the latest version of Pulumi and the of the Azure Provider.

**Comment by jen20** [comment]
Interestingly I cannot reproduce this in pure Terraform, despite the panic coming from inside the provider, so this must be a bug in how the Terraform bridge is working in this case.

**Comment by jen20** [comment]
This is however reproducible via Node and Python, so I think we can safely narrow down to a bug in the Terraform bridge (cc @swgillespie)

**Comment by jen20** [comment]
I've updated the repro for this in NodeJS:

```typescript
import * as azure from "@pulumi/azure";
import * as azuread from "@pulumi/azuread";

const keyMaterial = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC5skTnaa7ovvroA73vDPBamde8XT5AU1QwJi1CA1SXsHSDi+/CTFb59sImm4omiK2TMZ656xHIQvN8bL58BfQQTu7AFcfnlLUcIXmETZimu8ruP4R1+FfQCD2aKOdIb1VvP7p4IWTZArQ3HuKYkX4amfsLRDWGvTd6ST4HXTbopwHbiXhFpkRBsQJyKMACee/zCVRa7T+cK5QklcYW3HOtcqQwUROkAQjwnKA9xK2csG3rk99/GqerMVoUssjJeQiyiGPTtc9bTQ84zWc3KGYOojEAVUkMk8acdTAi0vlMeSf0Jng7atIauuR23qoAzW2pmCbARD0luDjpdIB0NFtL james@jen20.com";

const app = new azuread.Application("aks-app");

const sp = new azuread.ServicePrincipal("aks-sp", {
    applicationId: app.applicationId,
});

const spPwd = new azuread.ServicePrincipalPassword("aks-sppwd", {
    servicePrincipalId: sp.id,
    endDate: "2025-01-01T01:02:03Z",
    value: "password!CHANGEME123123#",
});

// Create an Azure Resource Group
const resourceGroup = new azure.core.ResourceGroup("aks-rg", {
    location: "WestUS",
});

const rgProps = {
    resourceGroupName: resourceGroup.name,
    location: resourceGroup.location,
};

const vnet = new azure.network.VirtualNetwork("aks-vnet", {
    ...rgProps,
    addressSpaces: ["10.0.0.0/8"],
});

const subnet = new azure.network.Subnet("aks-subnet", {
    resourceGroupName: resourceGroup.name,
    virtualNetworkName: vnet.name,
    addressPrefix: "10.0.0.0/23",
});

const subnetAssignment = new azure.role.Assignment("subnet-perms", {
    principalId: sp.id,
    roleDefinitionName: "Network Contributor",
    scope: subnet.id,
});

new azure.containerservice.KubernetesCluster("aks", {
    ...rgProps,
    dnsPrefix: "dns",
    agentPoolProfile: {
        name: "type1",
        count: 2,
        vmSize: "Standard_B2ms",
        osType: "Linux",
        maxPods: 110,
        vnetSubnetId: subnet.id,
    },
    linuxProfile: {
        adminUsername: "azureuser",
        sshKey: {
            keyData: keyMaterial,
        },
    },
    servicePrincipal: {
        clientId: app.applicationId,
        clientSecret: spPwd.value,
    },
    roleBasedAccessControl: {
        enabled: true,
    },
    networkProfile: {
        networkPlugin: "azure",
        serviceCidr: "10.10.0.0/16",
        dnsServiceIp: "10.10.0.10",
        dockerBridgeCidr: "172.17.0.1/16",
    }
}, {dependsOn: [subnetAssignment]});
```

The reason it fails when we update is because the Terraform bridge is passing the following value through:

```
"addon_profile.#":"1",
```

Without any actual values for `addon_profile.0`, causing the provider to fail here: https://github.com/terraform-providers/terraform-provider-azurerm/blob/002ea7e1ac7fdf061355bcb8b25445b54666179c/azurerm/resource_arm_kubernetes_cluster.go#L782

**Comment by jen20** [comment]
I've fixed this in `pulumi-terraform` and https://github.com/pulumi/pulumi-azure/pull/245 pulls it into `pulumi-azure`.

**Comment by jen20** [comment]
This was fixed in pulumi/pulumi-azure#245, and will ship with the next version of `pulumi-azure`. Thanks for the detailed repro, @4c74356b41!
